src/hamiltonian.jl

```diff
@@ -498,6 +498,7 @@ julia> similarmatrix(h, LinearAlgebraPackage()) |> summary
     `bloch!`
 """
 similarmatrix(h, dest_type = missing) = _similarmatrix(dest_type, matrixtype(h), h)
+similarmatrix(kets, dest_type, ketflatsize) = _similarmatrix(dest_type, typeof(kets), ketflatsize)
```

Comment:
This changes the API of similarmatrix, which is an exported function, see `?similarmatrix`. If we want `similarmatrix(kets, dest_type)`  to work for `kets::AbstractArray` the correct approach is to add `matrixtype` and `flatsize` methods that accept the AbstractArray